openstack/baremetalinspector/v1/introspection/requests.go

```diff
@@ -0,0 +1,115 @@
+func (opts ListIntrospectionsOpts) ToIntrospectionsListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// ListIntrospections makes a request against the Inspector API to list the current introspections.
+func ListIntrospections(client *gophercloud.ServiceClient, opts ListIntrospectionsOptsBuilder) pagination.Pager {
+	url := listIntrospectionsURL(client)
+	if opts != nil {
+		query, err := opts.ToIntrospectionsListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		var rpage = IntrospectionPage{pagination.LinkedPageBase{PageResult: r}}
+		return rpage
+	})
+}
+
+// GetIntrospectionStatus makes a request against the Inspector API to get the
+// status of a single introspection.
+func GetIntrospectionStatus(client *gophercloud.ServiceClient, NodeID string) (r GetIntrospectionStatusResult) {
+	_, r.Err = client.Get(introspectionURL(client, NodeID), &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// StartOptsBuilder ...
+type StartOptsBuilder interface {
+	ToStartIntrospectionQuery() (string, error)
+}
+
+// StartOpts ...
+type StartOpts struct {
+	// Whether the current installation of ironic-inspector can manage PXE booting of nodes.
+	ManageBoot bool `q:"manage_boot"`
+}
+
+// ToStartIntrospectionQuery ...
+func (opts StartOpts) ToStartIntrospectionQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// StartIntrospection initiate hardware introspection for node NodeID .
+// All power management configuration for this node needs to be done prior to calling the endpoint.
+func StartIntrospection(client *gophercloud.ServiceClient, NodeID string, opts StartOptsBuilder) (r StartResult) {
+	_, err := opts.ToStartIntrospectionQuery()
+	if err != nil {
+		r.Err = err
+		return
+	}
+
+	_, r.Err = client.Post(introspectionURL(client, NodeID), nil, nil, &gophercloud.RequestOpts{
+		OkCodes: []int{202},
+	})
+
+	return
+}
+
+// AbortIntrospection abort running introspection.
+func AbortIntrospection(client *gophercloud.ServiceClient, NodeID string) (r AbortResult) {
+	_, r.Err = client.Post(abortIntrospectionURL(client, NodeID), nil, nil, &gophercloud.RequestOpts{
+		OkCodes: []int{202},
+	})
+
+	return
+}
+
+// GetIntrospectionData return stored data from successful introspection.
+func GetIntrospectionData(client *gophercloud.ServiceClient, NodeID string) (r dataResult) {
+	_, r.Err = client.Get(introspectionDataURL(client, NodeID), &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// ApplyIntrospectionData triggers introspection on stored unprocessed data.
+// No data is allowed to be sent along with the request.
+func ApplyIntrospectionData(client *gophercloud.ServiceClient, NodeID string) (r ApplyDataResult) {
```

Comment:
Maybe `ReapplyIntrospection`? "Apply data" sounds a bit weird to me.

Reply:
changed